### DISCUSSION TITLE
query conditions for M2A field

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#17890] [Archive]
Hi Greetings,

I have a collection named clients, which has a M2A field named business. In the junction table "clients_business", there are a few fields automatically generated by Directus:
- id
- clients_id
- item
- collection

<img width="857" alt="image" src="https://user-images.githubusercontent.com/311397/227122150-55a1851c-2a72-4d31-ae43-f777428c248a.png">

I want to write a filter, to get the content of one specific business( business1 in below example), and for a specific client(abece82e-c8e3-4779-9920-7d0ce93488fd in below example)

```js
  const b = await directus.items("clients");
  const q = {
    limit: 1,
    fields: [`business.item:business1.profile`],
    filter: {
      _and: [
        {
          id: {
            _eq: "abece82e-c8e3-4779-9920-7d0ce93488fd",
          },
        },
        {
          business: {
            collection: {
              _eq: "business1",
            },
          },
        },
      ],
    },
  };
  const { data } = await b.readByQuery(q);
```

However, looks like the second part of the `_and` condition is not working, as I can see the result is regardlss the collection name, the items in other collection ( not in the business1 collection) is also returned.

Is there something wrong with the query object I drafted?

Thanks, Austin

**Comment by rijkvanzanten** [comment]
I believe you're caught in some confusion around "where" to filter. In this particular case, your filter says:

> "Give me 1 `clients` record where `id = <uid>` and `business has an item where collection = 'business1'`" 

What you're looking for (i think) is a filter that says:

> "Give me 1 `clients` record where `id = <uid>` and only nest the m2a items in `business` where `collection = 'business1'`" 

That's where the `deep` query param comes in. That applies you to apply any of the query params on the nested set of data. Try:

```js
const q = {
	limit: 1,
	fields: [`business.item:business1.profile`],
	filter: { id: { _eq: "abece82e-c8e3-4779-9920-7d0ce93488fd" } },
	deep: {
		business: {
			_filter: {
				collection: {
					_eq: "business1",
				},
			},
		},
	},
};
```

---

(Also fwiw, the sdk has a `readOne` utility method you could use to remove the need for `limit: 1` and `filter` on ID: `await directus.items('clients').readOne("abece82e-c8e3-4779-9920-7d0ce93488fd", q);`

**Reply by xu4wang** [reply]
@rijkvanzanten thanks for the clarification, and for helping me refactoring the code.

Below is the working query with readOne.

```js
export async function getClientProfile(
  client: string,
  profile: string
): Promise<number> {
  const b = await sdk_get_collection("clients");

  //Deep allows you to set any of the other query parameters on a nested relational dataset.  
  const q = {
    fields: [`business.item:${profile}.profile`],
    deep: {
      business: {
        _filter: {
          collection: {
            _eq: profile,
          },
        },
      },
    },
  };

  const  data  = await b.readOne(client, q);
  //console.log(JSON.stringify(data));
  if (data ) {
    return data.business[0].item.profile;
  } else {
    throw new Error(`Invalid client ID: ${client} or profile name ${profile}`);
  }
}

```
